packages/firebase_admob/README.md

```diff
@@ -182,6 +182,104 @@ RewardedVideoAd.instance.listener =
+as `MainActivity.java` as seen below:
+
+```java
+package my.app.path;
+
+import com.google.android.gms.ads.formats.UnifiedNativeAd;
+import com.google.android.gms.ads.formats.UnifiedNativeAdView;
+import io.flutter.plugins.firebaseadmob.FirebaseAdMobPlugin.NativeAdFactory;
+import java.util.Map;
+
+class NativeAdFactoryExample implements NativeAdFactory {
+  @Override
+  public UnifiedNativeAdView createNativeAd(
+      UnifiedNativeAd nativeAd, Map<String, Object> customOptions) {
+    // Create UnifiedNativeAdView
+  }
+}
+```
+ 
+An instance of a `NativeAdFactory` should also be added to the `FirebaseAdMobPlugin`. This is done
+differently whether you are using the Embedding V1 or Embedding V2.
+
+If you're using the Embedding V1, you need to retrieve the published `FirebaseAdMobPlugin` after
+calling `GeneratedPluginRegistrant.registerWith(this);`. Then, add the `NativeAdFactory` with a
+unique factory Id. You're
+`MainActivity.java` should look similar to:
+
+```java
+package my.app.path;
+
+import android.os.Bundle;
+import io.flutter.app.FlutterActivity;
+import io.flutter.plugins.GeneratedPluginRegistrant;
+import io.flutter.plugins.firebaseadmob.FirebaseAdMobPlugin;
+
+public class MainActivity extends FlutterActivity {
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    GeneratedPluginRegistrant.registerWith(this);
+
+    final FirebaseAdMobPlugin adMobPlugin =
+        valuePublishedByPlugin("io.flutter.plugins.firebaseadmob.FirebaseAdMobPlugin");
+    adMobPlugin.addNativeAdFactory("adFactoryExample", new NativeAdFactoryExample());
+  }
+}
+```
+
+If you're using the Embedding V2, you need to retrieve the published `FirebaseAdMobPlugin` from the
+`FlutterEngine`. Then, add the `NativeAdFactory` with a unique factoryId. You're `MainActivity.java`
+should look similar to:
+
+```java
+package my.app.path;
+
+import io.flutter.embedding.android.FlutterActivity;
+import io.flutter.embedding.engine.FlutterEngine;
+import io.flutter.plugins.firebaseadmob.FirebaseAdMobPlugin;
+
+public class MainActivity extends FlutterActivity {
+  @Override
+  public void configureFlutterEngine(FlutterEngine flutterEngine) {
+    flutterEngine.getPlugins().add(new FirebaseAdMobPlugin());
+
+    final FirebaseAdMobPlugin adMobPlugin =
+        (FirebaseAdMobPlugin) flutterEngine.getPlugins().get(FirebaseAdMobPlugin.class);
+    adMobPlugin.addNativeAdFactory("adFactoryExample", NativeAdFactoryExample());
+  }
+}
+```
+
+When creating the `NativeAd` in Flutter, the `factoryId` parameter should match the one you used to
+add the factory to `FirebaseAdMobPlugin`.
+
+An example of displaying a `UnifiedNativeAd` with a `UnifiedNativeAdView` can be found
+[here](https://developers.google.com/admob/android/native/advanced). The example app also inflates
+a custom layout and displays the test Native ad.
+
```

Comment:
Would be useful to add some explanation about how to use the Dart-side API

Reply:
done